gtubt/lib/ui/blocs/appbar_bloc/appbar_bloc.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:GTUBT/service/user.dart';
+import 'package:GTUBT/ui/blocs/appbar_bloc/appbar_event.dart';
+import 'package:GTUBT/ui/blocs/appbar_bloc/appbar_state.dart';
+import 'package:bloc/bloc.dart';
+
+class AppbarBloc extends Bloc<AppbarEvent, AppbarState> {
+  UserService userService = UserService();
+  @override
+  AppbarState get initialState => AppbarState();
+
+  @override
+  Stream<AppbarState> mapEventToState(AppbarEvent event) async* {
+    if (event is UserEditButtonPressed) {
+      if (this.state.editMode) {
+        userService.patch(userService.currentUser);
+      }
+
+      yield AppbarState(editMode: !getCurrentEditButtonState());
+    } else if (event is PageChangedAppbarEvent) {
+      yield AppbarState();
+    }
+  }
+
+  bool getCurrentEditButtonState() {
```

Comment:
buna neden ihtiyac duyuyoruz? `BlocProvider.of<AppbarBloc>(context).state` diyerek ulasabiliyoruz state durumuna.

Reply:
Basta bunun icinde baska checkler de vardi. O yuzden fonksiyona cevirmistim kalmis boyle. bahsettigin kullanima geri ceviririm.